src/Middleware/AuthenticationMiddleware.php

```diff
@@ -158,6 +158,12 @@ protected function getRedirectUrl($target, ServerRequestInterface $request)
             $query = '?' . $query;
         }
 
+        $hashPos = strpos($target, '#');
+        if ($hashPos !== false) {
+            $query = $query . substr($target, $hashPos);
```

Comment:
This could be

    $query .= substr($target, $hashPos);

Reply:
Commited. Didn't shortened for readability. Guess could be shortened even further.